.github/workflows/ci.yml

```diff
@@ -26,26 +42,103 @@ jobs:
 
       - name: hash test inputs
         run: |
           if ! which sha256sum; then brew install coreutils; fi
           sha256sum $(find test-vectors -name '*.json') > test-vector-hashes.txt
 
-      - name: Run Gradle Tasks
-        run: gradle build koverXmlReport
+      - name: Build, Test, and Verify
+        run: |
+          # Maven "verify" lifecycle will build, test, and verify
+          mvn verify \
+              --batch-mode \
+              -P sign-artifacts
+        env:
+          SIGN_KEY_PASS: ${{ secrets.GPG_SECRET_PASSPHRASE }}
+          SIGN_KEY: ${{ secrets.GPG_SECRET_KEY }}
 
+  # On Ubuntu we build, test, verify, and deploy: Code Coverage, Test Vectors, and SNAPSHOT artifacts to TBD Artifactory
+  build-test-deploy-snapshot-ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # https://cashapp.github.io/hermit/usage/ci/
+      - name: Init Hermit
+        uses: cashapp/activate-hermit@v1
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: hash test inputs
+        run: |
+          if ! which sha256sum; then brew install coreutils; fi
+          sha256sum $(find test-vectors -name '*.json') > test-vector-hashes.txt
+
+      - name: Resolve Snapshot Version
+        id: resolve_version
+        run: |
+          # Version resolution: use provided
+          if [ -n "${{ github.event.inputs.version }}" ]; then
+            resolvedVersion=${{ github.event.inputs.version }}
+          # Otherwise, construct a version for deployment in form X.Y.Z-commit-$shortSHA-SNAPSHOT
+          else
+            longSHA=$(git rev-parse --verify HEAD)
+            shortSHA=$(echo "${longSHA:0:7}")
+            resolvedVersion="commit-$shortSHA-SNAPSHOT"
+            echo "Requesting deployment as version: $resolvedVersion"
+          fi
+
+          # Postcondition check; only allow this to proceed if we have a version ending in "-SNAPSHOT"
+          if [[ ! "$resolvedVersion" =~ -SNAPSHOT$ ]]; then
+            echo "Error: The version does not end with \"-SNAPSHOT\": $resolvedVersion"
+            exit 1
+          fi
+
+          echo "Resolved SNAPSHOT Version: $resolvedVersion"
+          echo "resolved_version=$resolvedVersion" >> $GITHUB_OUTPUT
+
+      - name: Build, Test, and Deploy to TBD Artifactory
+        run: |
+          # Set newly resolved version in POM config
+          mvn \
+            versions:set \
+            --batch-mode \
+            -DnewVersion=${{ steps.resolve_version.outputs.resolved_version }}
+
+          # Maven deploy lifecycle will build, run tests, verify, sign, and deploy
+          mvn \
+            deploy \
+              --batch-mode \
+              --settings .maven_settings.xml \
+              -P sign-artifacts
```

Comment:
chad tells me sign-artifacts is a custom profile. can you point to where that is and what it does?

Reply:
Yep it's in the root/parent `pom.xml` and it does GPG signing of the artifacts using the TBD key stored in secrets.

Reply:
The workflow sets the env vars as expected by the Maven Plugin doing the signing: https://www.simplify4u.org/sign-maven-plugin/sign-mojo.html